consumer/consumer.go

```diff
@@ -163,23 +168,31 @@ func (c *Consumer) handleMessages(ch <-chan *sarama.ConsumerMessage, offset offs
 		for {
 			attempts++
 
-			m := c.convertMessage(msg)
 			// Note: The second returned value is not checked because we will never receive messages for a topic
 			// that it does not have a handler for.
 			h, _ := c.handlers.Get(msg.Topic)
-			err := h.(handler.Handler).HandleMessage(m)
+
+			// if an error occurs during an unformat call, it usually means
+			// the chosen converter was the wrong one. the consumption must block
+			// and an error must be reported.
+			m, err := h.Converter.FromKafka(*c.config, msg)
 			if err == nil {
-				offset.MarkOffset(msg, "")
-				if c.Metrics != nil {
-					c.Metrics.Report(*m, map[string]string{
-						"attempts":        strconv.Itoa(attempts),
-						"msgOffset":       strconv.FormatInt(msg.Offset, 10),
-						"remainingOffset": strconv.FormatInt(max.HighWaterMarkOffset()-msg.Offset, 10),
-					})
+				err = h.Handler.HandleMessage(m)
+				if err == nil {
+					offset.MarkOffset(msg, "")
+					if c.Metrics != nil {
+						c.Metrics.Report(*m, map[string]string{
+							"attempts":        strconv.Itoa(attempts),
+							"msgOffset":       strconv.FormatInt(msg.Offset, 10),
+							"remainingOffset": strconv.FormatInt(max.HighWaterMarkOffset()-msg.Offset, 10),
+						})
+					}
+					break
 				}
-				break
 			}
 
+			c.Logger.Printf("an error occured while consuming a message: %v, retrying after %s.", err, c.config.RetryInterval)
```

Comment:
```suggestion
			c.Logger.Printf("an error occured while consuming a message: %v, retrying after %s second(s).", err, c.config.RetryInterval)
```

Reply:
No need, `RetryInterval` is a time.Duration, which is a `fmt.Stringer`: automatically add the right unit at the end of the string.